launch/launcher.go

```diff
@@ -124,6 +146,85 @@ func (l *Launcher) env(process Process) error {
 	})
 }
 
+func (l *Launcher) execD(process Process) error {
+	execDs, err := l.buildpackFiles(process, ExecDDirName, supportsExecD)
+	if err != nil {
+		return errors.Wrapf(err, "failed to find all exec.d executables in layers dir, '%s'", l.LayersDir)
+	}
+	for _, execD := range execDs {
+		if err := l.ExecD.ExecD(execD, l.Env); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func supportsExecD(bp Buildpack) bool {
+	if bp.API == "" {
+		return false
+	}
+	return api.MustParse(bp.API).Compare(api.MustParse("0.5")) >= 0
+}
+
+func (l *Launcher) buildpackFiles(process Process, dirName string, filters ...func(Buildpack) bool) ([]string, error) {
+	var files []string
+
+	appendIfFile := func(path string, fi os.FileInfo) {
+		if !fi.IsDir() {
+			files = append(files, path)
+		}
+	}
+
+	appendFilesInDir := func(path string) error {
+		fis, err := ioutil.ReadDir(path)
+		if os.IsNotExist(err) {
+			return nil
+		}
+		if err != nil {
+			return errors.Wrapf(err, "failed to list files in dir '%s'", path)
+		}
+
+		for _, fi := range fis {
+			appendIfFile(filepath.Join(path, fi.Name()), fi)
+		}
+		return nil
+	}
+
+	if err := l.eachBuildpack(func(path string) error {
+		absPath, err := filepath.Abs(path)
+		if err != nil {
+			return err
+		}
+		return eachDir(absPath, func(path string) error {
+			if err := appendFilesInDir(filepath.Join(path, dirName)); err != nil {
+				return err
+			}
+			if process.Type != "" {
+				return appendFilesInDir(filepath.Join(path, dirName, process.Type))
+			}
+			return nil
+		})
+	}, filters...); err != nil {
+		return nil, err
```

Comment:
Because of this line https://github.com/buildpacks/lifecycle/pull/473/files#diff-dcbaa95f60b04e4252f4de7f9e6718e4f63b3f5e516a8871b6e9da20d2c51461R152, the error would end up saying `failed to find all exec.d executables in layers dir` twice if I accepted this suggestion. Also this helper method is used for profile scripts as well.